gdb-server/src/target/base.rs

```diff
@@ -21,7 +21,12 @@ impl MultiThreadBase for RuntimeTarget<'_> {
         let mut core = session.core(tid.get() - 1).into_target_result()?;
 
         regs.pc = core
-            .read_core_reg(core.registers().program_counter())
+            .read_core_reg(
+                core.registers()
+                    .program_counter()
+                    .into_target_result()?
+                    .id(),
```

Comment:
Can we make it so this is an `Into` implementation that does this implicitely?

Reply:
@Yatekii As previously mentioned, there is already an `Into` implementation in `core::registers`, but apparently Rust won't invoke it implicitly after the `?`. You can do it explicitly by adding a `.into()`. I'm going to settle for just adding `.id()` because I find it easier to read. 

If you discover a way to force the implicit `Into`, I'd be happy to implement it :)